waitext/extensions.go

```diff
@@ -0,0 +1,118 @@
+package waitext
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	DefaultInitialInterval     = 500 * time.Millisecond
+	DefaultRandomizationFactor = 0.5
+	DefaultMultiplier          = 1.5
+	DefaultMaxInterval         = 60 * time.Second
+
+	// Forever should be used to simulate infinite retries or backoff increase.
+	// Usually it's wise to have a context with timeout to avoid an infinite loop.
+	Forever = math.MaxInt32
+)
+
+// NewExponentialBackoff creates a backoff that increases the delay between each step based on a factor.
+// If maxInterval is positive, then the wait duration will not exceed its value while increasing.
+// Essentially at step N the wait is min(initialInterval*factor^(N-1), maxInterval)
+func NewExponentialBackoff(initialInterval time.Duration, factor float64, maxInterval time.Duration) wait.Backoff {
```

Comment:
It seems like this function is not used outside the package. Would it be okay to delete it?